flexmeasures/data/models/time_series.py

```diff
@@ -682,3 +683,83 @@ def parse_source_arg(
         else:
             parsed_sources.append(source)
     return parsed_sources
+
+
+class Annotation(db.Model):
+    """An annotation is a nominal value that applies to a specific time or time span.
+
+    Examples of annotation types:
+        - user annotation: annotation.type == "label" and annotation.source.type == "user"
+        - unresolved alert: annotation.type == "alert"
+        - resolved alert: annotation.type == "label" and annotation.source.type == "alerting script"
+        - organisation holiday: annotation.type == "holiday" and annotation.source.type == "user"
+        - public holiday: annotation.type == "holiday" and annotation.source.name == "workalendar"
+    """
+
+    id = db.Column(db.Integer, nullable=False, autoincrement=True, primary_key=True)
+    name = db.Column(db.String(255), nullable=False)
+    start = db.Column(db.DateTime(timezone=True), nullable=False)
+    end = db.Column(db.DateTime(timezone=True), nullable=False)
+    source_id = db.Column(db.Integer, db.ForeignKey(DataSource.__tablename__ + ".id"))
+    source = db.relationship(
+        "DataSource",
+        foreign_keys=[source_id],
+        backref=db.backref("annotations", lazy=True),
+    )
+    type = db.Column(db.Enum("alert", "holiday", "label", name="annotation_type"))
+    UniqueConstraint("name", "start", "source_id", "type")
```

Comment:
This looks like a weird way to add a constraint. I wonder why this seems to have worked. I used the `__table_args__`, which I found was recommended by SqlAlchemy (see `GenericAsset`)